process/headerCheck/headerIntegrityVerifier.go

```diff
@@ -0,0 +1,59 @@
+package headerCheck
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data"
+)
+
+type headerIntegrityVerifier struct {
+}
+
+// NewHeaderIntegrityVerifier returns a new instance of headerIntegrityVerifier
+func NewHeaderIntegrityVerifier() *headerIntegrityVerifier {
+	return &headerIntegrityVerifier{}
```

Comment:
save chain ID than you do not need to get the reference

Reply:
done